net/Socket.cpp

```diff
@@ -1145,63 +1158,160 @@ bool ServerSocket::bind([[maybe_unused]] Type type, [[maybe_unused]] int port)
-    try
+    if (rc < 0)
     {
-        // Create a socket object using the factory.
-        if (rc != -1)
+        const int cause = errno;
+        constexpr const char * messagePrefix = "Failed to accept. (errno: ";
+        if (!isRecoverableAcceptError(cause))
         {
-#if !MOBILEAPP
-            char addrstr[INET6_ADDRSTRLEN];
-
-            Socket::Type type;
-            const void *inAddr;
-            if (clientInfo.sin6_family == AF_INET)
-            {
-                struct sockaddr_in *ipv4 = (struct sockaddr_in *)&clientInfo;
-                inAddr = &(ipv4->sin_addr);
-                type = Socket::Type::IPv4;
-            }
+            // Unrecoverable error, forceExit instead of throw (closing `ServerSocket` and discard it from `AcceptPoll` during `SocketPoll::poll`)
+            LOG_FTL(messagePrefix << std::to_string(cause) << ", " << std::strerror(cause) << ')');
+#if TEST_SERVERSOCKET_FAULT_INJECTION
+            if (injectedError)
+                throw std::runtime_error(std::string(messagePrefix) + std::to_string(cause) + ", " + std::strerror(cause) + ')');
             else
-            {
-                struct sockaddr_in6 *ipv6 = &clientInfo;
-                inAddr = &(ipv6->sin6_addr);
-                type = Socket::Type::IPv6;
-            }
+#endif
+            Util::forcedExit(EX_SOFTWARE);
+        }
+        else // Recoverable error, ignore to retry
+            LOG_DBG(messagePrefix << std::to_string(cause) << ", " << std::strerror(cause) << ')');
+        return nullptr;
+    }
+    LOG_TRC("Accepted socket #" << rc << ", creating socket object.");
 
-            std::shared_ptr<Socket> _socket = createSocketFromAccept(rc, type);
+#if !MOBILEAPP
+    char addrstr[INET6_ADDRSTRLEN];
 
-            ::inet_ntop(clientInfo.sin6_family, inAddr, addrstr, sizeof(addrstr));
-            _socket->setClientAddress(addrstr, clientInfo.sin6_port);
+    Socket::Type type;
+    const void *inAddr;
+    if (clientInfo.sin6_family == AF_INET)
+    {
+        struct sockaddr_in *ipv4 = (struct sockaddr_in *)&clientInfo;
+        inAddr = &(ipv4->sin_addr);
+        type = Socket::Type::IPv4;
+    }
+    else
+    {
+        struct sockaddr_in6 *ipv6 = &clientInfo;
+        inAddr = &(ipv6->sin6_addr);
+        type = Socket::Type::IPv6;
+    }
+    ::inet_ntop(clientInfo.sin6_family, inAddr, addrstr, sizeof(addrstr));
 
-            LOG_TRC("Accepted socket #" << _socket->getFD() << " has family "
-                                        << clientInfo.sin6_family << ", " << *_socket);
-#else
-            std::shared_ptr<Socket> _socket = createSocketFromAccept(rc, Socket::Type::Unix);
-#endif
-            return _socket;
-        }
-        return std::shared_ptr<Socket>(nullptr);
+    const size_t extConnCount = StreamSocket::getExternalConnectionCount();
+    if( 0 < net::Defaults.maxExtConnections && extConnCount >= net::Defaults.maxExtConnections )
```

Comment:
Whitespace if(<space>foo) doesn't fit with surrounding code - please don't use it.

Reply:
OK